server/src/main/java/org/elasticsearch/health/node/LocalHealthMonitor.java

```diff
@@ -112,42 +135,80 @@ private void maybeScheduleNextRun(TimeValue time) {
     }
 
     // Helper method that starts the monitoring without a delay.
-    private void maybeScheduleNow() {
+    // Visible for testing
+    void maybeScheduleNow() {
         maybeScheduleNextRun(TimeValue.ZERO);
     }
 
     @Override
     public void clusterChanged(ClusterChangedEvent event) {
-        if (prerequisitesFulfilled == false) {
-            prerequisitesFulfilled = event.state().nodesIfRecovered().getMinNodeVersion().onOrAfter(Version.V_8_5_0)
-                && HealthMetadata.getFromClusterState(event.state()) != null;
-            maybeScheduleNow();
+        DiscoveryNode current = HealthNode.findHealthNode(event.state());
+        prerequisitesFulfilled = event.state().nodesIfRecovered().getMinNodeVersion().onOrAfter(Version.V_8_5_0)
+            && HealthMetadata.getFromClusterState(event.state()) != null
+            && current != null;
+        if (prerequisitesFulfilled) {
+            DiscoveryNode previous = HealthNode.findHealthNode(event.previousState());
```

Comment:
nit: maybe on the naming bit `previousHealthNode`? 

IMO it'd make the following lines a bit more readable where we "talk" about previous/current

Reply:
As before, shall we shortcircuit this branch with a null check for `previous` ? (for better visibility that it _can_ be null)

Reply:
I do not see what we could shortcircuit here.... Can you be more specific?